strawberry/subscriptions/protocols/graphql_transport_ws/handlers.py

```diff
@@ -217,6 +217,11 @@ async def operation_task(
             await self.handle_async_results(result_source, operation_id)
         except BaseException:  # pragma: no cover
             # cleanup in case of something really unexpected
+            # wait for generator to be closed to ensure that any existing
+            # 'finally' statement is called
+            generator = self.subscriptions[operation_id]
+            with suppress(RuntimeError):
+                await generator.aclose()
```

Comment:
```suggestion
            with suppress(RuntimeError):
                await result_source.aclose()
```

`self.subscriptions[operation_id]` is the same as `result_source`.

Reply:
Renamed generator in commit: https://github.com/strawberry-graphql/strawberry/pull/2141/commits/0f70ef209bc178def787118d7088ec1b5165a542